pkg/skaffold/deploy/cloudrun/deploy.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2022 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cloudrun
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/option"
+	"google.golang.org/api/run/v1"
+	k8syaml "sigs.k8s.io/yaml"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/access"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/label"
+	deploy "github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/types"
+	sErrors "github.com/GoogleContainerTools/skaffold/pkg/skaffold/errors"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/gcp"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/manifest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/log"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/status"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/sync"
+	"github.com/GoogleContainerTools/skaffold/proto/v1"
+)
+
+// Deployer deploys code to Google Cloud Run.
+type Deployer struct {
+	logger   log.Logger
+	monitor  *Monitor
+	labeller *label.DefaultLabeller
+
+	DefaultProject string
+	Region         string
+
+	Cfg deploy.Config
+
+	// additional client options for connecting to Cloud Run, used for tests
+	clientOptions []option.ClientOption
+	useGcpOptions bool
+}
+
+// NewDeployer creates a new Deployer for Cloud Run from the Skaffold deploy config.
+func NewDeployer(labeller *label.DefaultLabeller, crDeploy *latest.CloudRunDeploy) (*Deployer, error) {
+	return &Deployer{
+		DefaultProject: crDeploy.DefaultProjectID,
+		Region:         crDeploy.Region,
+		logger:         &log.NoopLogger{},
```

Comment:
nit: Is it possible to stream the logs from the cloud run deployment and add that as a logger here?  Perhaps not in this PR but as a TODO, I think it would be valuable for users familiar w/ skaffolds log streaming (perhaps even just fetching the link to the logs and displaying it)

Reply:
Awesome, if the roadmap of TODOs is ok to be public it might be helpful to create `Issues` for some of the future items so the Skaffold team can understand the roadmap and possibly track features in milestones